### ISSUE TITLE
How to use array of empirical data as function coefficients?

### ISSUE THREAD

**Issue by mattragoza** [FEniCS/dolfinx#2315]
Hello, I am working on a parameter identification problem where I have measured displacement data on a uniform grid (in a numpy array) and I would like to solve a PDE to identify the spatially-varying stiffness of the material.

How can I assign the measured data values to the coefficients of a function in the finite element space?

Here is what I am trying and where I am getting stuck:

```python
# u is a numpy array containing measured displacement data
n_x, n_y = u.shape
resolution = 1e-3

# spatial coordinates where u is observed
x = np.stack(np.meshgrid(np.arange(n_x), np.arange(n_y)), axis=-1) * resolution
print(x)
```
Produces the output:
```
array([[0.   , 0.   ],
       [0.   , 0.001],
       [0.   , 0.002],
       ...,
       [0.079, 0.097],
       [0.079, 0.098],
       [0.079, 0.099]])
```

When I try to create a rectangular mesh:

```python
x_min = x.min(axis=0)
x_max = x.max(axis=0)

mesh = dolfinx.mesh.create_rectangle(
    comm=MPI.COMM_WORLD,
    points=[x.min(axis=0), x.max(axis=0)], n=[n_x - 1, n_y - 1],
    cell_type=dolfinx.mesh.CellType.triangle
)
print(mesh.geometry.x[:,:2]) # only 2D values are nonzero
```
The order of the coordinates is different:
```
array([[0.078, 0.   ],
       [0.079, 0.   ],
       [0.079, 0.001],
       ...,
       [0.   , 0.098],
       [0.001, 0.099],
       [0.   , 0.099]])
```
In summary, what I am trying to do is something like this:
```python
V = dolfinx.fem.FunctionSpace(mesh, ('Lagrange', 1))
u_h = dolfinx.fem.Function(V)
u_h.interpolate(u_array) # this doesn't work
```
But this fails, because interpolate expects a function object, not an array. How can I specify the coefficients on a set of predefined nodes from data?

**Comment by nate-sime** [comment]
This is not an issue with dolfinx development, rather it is a user specific numerical scheme and implementation. Consider posting your question to [the discourse forum](https://fenicsproject.discourse.group/).
